dev_guide/builds/build_inputs.adoc

```diff
@@ -0,0 +1,833 @@
+`BuildConfig.spec.source.dockerfile` field is specified, the inline Dockerfile
+replaces the one in the `contextDir` of the Git repository.
+
+The source definition is part of the `spec` section in the `BuildConfig`:
+
+[source,yaml]
+----
+source:
+  type: "Git"
+  git: <1>
+    uri: "https://github.com/openshift/ruby-hello-world"
+    ref: "master"
+  contextDir: "app/dir" <2>
+  dockerfile: "FROM openshift/ruby-22-centos7\nUSER example" <3>
+----
+<1> The `git` field contains the URI to the remote Git repository of the
+source code. Optionally, specify the `ref` field to check out a specific Git
+reference. A valid `ref` can be a SHA1 tag or a branch name.
+<2> The `contextDir` field allows you to override the default location inside
+the source code repository where the build looks for the application source
+code. If your application exists inside a sub-directory, you can override the
+default location (the root folder) using this field.
+<3> If the optional `dockerfile` field is provided, it should be a string
+containing a Dockerfile that overwrites any Dockerfile that may exist in the
+source repository.
+
+When using the Git repository as a source without specifying the `ref`
+field, {product-title} performs a shallow clone (`--depth=1` clone). That means
+only the `HEAD` (usually the `master` branch) is downloaded. This results
+in repositories downloading faster, including the commit history.
+
+A shallow clone is also used when the `ref` field is specified and set to an
+existing remote branch name. However, if you specify the `ref` field to a
+specific commit, the system will fallback to a regular Git clone operation and
+checkout the commit, because using the `--depth=1` option only works with named
+branch refs.
+
+To perform a full Git clone of the `master` for the specified repository, set
+the `ref` to `master`.
+
+[[using-a-proxy]]
+=== Using a Proxy
+
+// tag::using-a-proxy-for-git-cloning-1[]
+
+If your Git repository can only be accessed using a proxy, you can define the
+proxy to use in the `source` section of the `BuildConfig`. You can configure
+both a HTTP and HTTPS proxy to use. Both fields are optional.  Domains for which
+no proxying should be performed can also be specified via the *NoProxy* field.
+
+[NOTE]
+====
+Your source URI must use the HTTP or HTTPS protocol for this to work.
+====
+
+[source,yaml]
+----
+source:
+  type: Git
+  git:
+    uri: "https://github.com/openshift/ruby-hello-world"
+    httpProxy: http://proxy.example.com
+    httpsProxy: https://proxy.example.com
+    noProxy: somedomain.com, otherdomain.com
+----
+// end::using-a-proxy-for-git-cloning-1[]
+
+ifdef::openshift-enterprise,openshift-origin[]
+Cluster administrators can also
+xref:../../install_config/http_proxies.adoc#configuring-hosts-for-proxies-using-ansible[configure
+a global proxy for Git cloning using Ansible].
+endif::[]
+
+[[source-clone-secrets]]
+=== Source Clone Secrets
```

Comment:
clone is reasonable, that's really what we're using it for...the git clone operation.  it's just a bit confusing since the api doesn't mention "clone".